### PR TITLE
Converted vmware-csi-driver-reqs snippet into module

### LINKED ISSUE #43297: This is a module.
The ban on xrefs in modules extends to snippets included in modules. The xref in this snippet needs to be removed, and this snippet needs to be changed into a module and added the appropriate assemblies, : https://github.com/openshift/openshift-docs/commit/33c01c0561c75a76587406e3a1ef2fd364df2378#diff-b9ebb27493b5768232354fdd741c985f20818ee281bd5491d9a5c1e881bf8d19R95

@bscott-rh

### PR DESCRIPTION
**Pull request by bscott-rh**
4.10+

Fixes #43297
[Preview of storage section](https://deploy-preview-43407--osdocs.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-vsphere.html#vsphere-csi-driver-reqs_persistent-storage-csi-vsphere)
[Preview of vSphere section](https://deploy-preview-43407--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-restricted-networks-vsphere.html#vsphere-csi-driver-reqs_installing-restricted-networks-vsphere)
[Preview of VMC section](https://deploy-preview-43407--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vmc/installing-vmc-network-customizations-user-infra.html#vsphere-csi-driver-reqs_installing-vmc-network-customizations-user-infra)

This change converts a snippet into a module. It also removes an xref from the snippet and places it in the assemblies that reference the module.

### CODE REVIEW

File: installing/installing_vmc/installing-restricted-networks-vmc-user-infra.adoc [hunk 1]
```diff
@@ -38,6 +38,9 @@ include::modules/cluster-entitlements.adoc[leveloffset=+1]
 
 include::modules/installation-vsphere-infrastructure.adoc[leveloffset=+1]
 
+include::modules/vmware-csi-driver-reqs.adoc[leveloffset=+1]
+For more information about removing a third-party CSI driver, see xref:../../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-install-issues_persistent-storage-csi-vsphere[Removing a third-party vSphere CSI Operator Driver].
+
 [role="_additional-resources"]
```
Comment (kalexand-rh) [inline installing/installing_vmc/installing-restricted-networks-vmc-user-infra.adoc:44]: Is this additional resources section still placed correctly? This link would be better in an additional resources section.

**Review by kalexand-rh — COMMENTED** [review]
I have a question, but this otherwise looks good. Thanks!

File: installing/installing_vmc/installing-restricted-networks-vmc-user-infra.adoc (hunk 1 above)
Comment (bscott-rh) [reply installing/installing_vmc/installing-restricted-networks-vmc-user-infra.adoc:43]: I agree - I moved the xref to the additional resources section.

File: storage/container_storage_interface/persistent-storage-csi-vsphere.adoc [hunk 2]
```diff
@@ -36,8 +36,8 @@ include::modules/persistent-storage-csi-about.adoc[leveloffset=+1]
 
 include::modules/persistent-storage-csi-vsphere-stor-policy.adoc[leveloffset=+1]
 
-[id="vsphere-csi-driver-requirements_{context}"]
-include::snippets/vmware-csi-driver-reqs.adoc[leveloffset=+1]
+include::modules/vmware-csi-driver-reqs.adoc[leveloffset=+1]
+For more information about removing a third-party CSI driver, see xref:../../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-install-issues_persistent-storage-csi-vsphere[Removing a third-party vSphere CSI Operator Driver].
```
Comment (kalexand-rh) [inline storage/container_storage_interface/persistent-storage-csi-vsphere.adoc:40]: Same here - this needs to be in an additional resources section.

**Comment by bscott-rh** [comment]
@racedo @julienlim I updated the infrastructure requirements to state that 6.7U3 is deprecated in 4.10 and reworded it slightly to make it clear that it's supported in 4.10 but will not be supported in 4.11. Can you take a look? [Preview link is here](https://deploy-preview-43407--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#installation-vsphere-infrastructure_installing-vsphere-installer-provisioned)

File: storage/container_storage_interface/persistent-storage-csi-vsphere.adoc (hunk 2 above)
Comment (bscott-rh) [reply storage/container_storage_interface/persistent-storage-csi-vsphere.adoc]: Fixed

**Comment by julienlim** [comment]
PR changed midstream so will wait till t he new PR for the version change.

**Pull request merged**